wgengine/netstack/netstack.go

```diff
@@ -793,3 +796,56 @@ func stringifyTEI(tei stack.TransportEndpointID) string {
 func ipPortOfNetstackAddr(a tcpip.Address, port uint16) (ipp netaddr.IPPort, ok bool) {
 	return netaddr.FromStdAddr(net.IP(a), int(port), "") // TODO(bradfitz): can do without allocs
 }
+
+// removeECNBits modifies p to remove ECN bits, if any.
+func removeECNBits(p *packet.Parsed) {
+	if p.IPVersion == 0 {
```

Comment:
I don't understand why this is here. It is harmless, just can't think of cases where it would be needed.

Reply:
It's a necessary precondition to accessing the packet bytes. We're only guaranteed to have a TCP+IP header in the buffer if we successfully parsed it as some kind of IP plus TCP.